### ISSUE TITLE
the page frozen in `Claiming rewards`

### ISSUE THREAD

**Issue by LE0x3L** [pk910/PoWFaucet#1]
Hi... Is the second time today that the page has not transferred the GoETH... still in `Claiming rewards` but nothing happened... I have attached a screenshot... what happened with my rewards?... thanks

![2022-08-16-175953_1515x1059_scrot](https://user-images.githubusercontent.com/1672925/185000014-8892d1ff-32bf-479d-b51d-d311c47aaca3.png)

**Comment by pk910** [comment]
Hi @LE0x3L,
Thanks for your report 👍

I've looked into the logs and found 2 sessions for your address that haven't been claimed and timed out.
I'm not sure what happened there... 
Could it be, that your internet connection gets interrupted sometimes?
The page tries to close the session gracefully 30 seconds before it times out, so it's not killed from server side... 
It looks like your connection was interrupted during this time and the session timed out.

I'll try to improve the implementation so timed out sessions are not lost and can still be claimed.
For the meantime Try to avoid mining the full 3h and click on "Close Session & Claim Rewards" a few minutes before it times out. That way you still have time left to reload the page if anything goes wrong.

I've sent you the missed rewards (7.81 + 6.545) from your sessions on goerli manually :)

**Comment by pk910** [comment]
Heya,

I've improved the claim logic a bit to handle timeouts better.
That should fix your issue. Can you retry?

**Comment by LE0x3L** [comment]
Hi @pk910 . . . I'm sorry for the delay . . . thanks for my missed rewards

The last week I tested and it failed again... when to reach any limit (session time or Maximum Claim Reward) the system did not transfer the reward ( I lost it again :'( ) . . . I'm using Firefox, I don't know if this matters. . . regards

![2022-08-17-193103_1920x1080_scrot](https://user-images.githubusercontent.com/1672925/185955976-1743cebe-5354-4d5e-bb33-fa395f7a94ab.png)

**Comment by LE0x3L** [comment]
some minutes ago, I've has the same result... I don´t reach out to claim the rewards before the session expires and I lost it :/

**Comment by pk910** [comment]
Hi @LE0x3L 

Sorry for the delay from my side, too.. I think I've closed the issue too early - haven't seen your answer,,

Thanks for coming back. I'm not fully sure about the reason for your issue, but I'll try to analyze it more closely in the next days 👍

**Comment by pk910** [comment]
Hmm, your issue should have been fixed by the patch above (1a612a466223557dfb899750ce6ac6bcddccd2a2). 
Your client should have received a claim token along with the "Session timeout" error message IF your session reached the minimal claimable amount (0.05 GöETH currently).

If your mining balance was above 0.05 GöETH and you still haven't been able to claim the only way I can think of is that your browser cached an older version of the faucet js files and therefore ignored the claim token... 
I've noticed this caching issue recently and it should be fixed with 16c8787b9d6bb30fe2ac7de0cfe2ddea6c7bb698 now.

The faucet page now also shows the currently loaded version of the client script..
It should show "v1.0.11" (or higher) in the bottom right corner of the page or show a warning when running a older version. 

May I ask you to try again?
